src/App.tsx

```diff
@@ -1,26 +1,58 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from './components/Error/ErrorNotification';
+import { TodoList } from './components/TodoList/TodoList';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoCount } from './components/TodoCount/TodoCount';
+import { TodoFilter } from './components/Filter/TodoFilter';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortBy, setSortBy] = useState('all');
+  const [errorMessage, setErrorMessage] = useState('');
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    getTodos(user?.id || 0)
+      .then(todosFromServer => setTodos(todosFromServer))
```

Comment:
Let’s use `async await`. It gives you a simpler way to work with asynchronous promise-based code